Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcess.java

```diff
@@ -157,8 +164,40 @@ public void stackFrameChanged() {
     myDASExecutionContextId = dasExecutionContextId;
 
     if (DebugType.REMOTE == debugType) {
-      // TODO won't work since Dart SDK 1.22 because auth token in URL is required
-      scheduleConnect("ws://" + myDebuggingHost + ":" + myObservatoryPort + "/ws");
+      final String httpDebuggerUrl =
+        Messages.showInputDialog(getSession().getProject(), "Enter a URL to a running Dart or Flutter application",
+                                 "Connect to a Running App", null, "http://127.0.0.1:12345/AUTH_CODE=/", new InputValidator() {
+            @Override
+            public boolean checkInput(final String inputString) {
+              return isDebugUrl.test(inputString);
+            }
+
+            @Override
+            public boolean canClose(String inputString) {
+              return true;
+            }
+          });
+      if (isDebugUrl.test(httpDebuggerUrl)) {
+        assert httpDebuggerUrl != null;
+        final String wsToConnect;
+        if (httpDebuggerUrl.startsWith("http://")) {
```

Comment:
you might want to first trim the URL. We noticed DevTools users had trouble copying the URLs with leading and trailing spaces and so we stripped the spaces to avoid confusing failure to connect errors.

Reply:
Good to know, PR updated.  Thanks Jacob